tests/test_python3.py

```diff
@@ -63,3 +69,59 @@ def test_python3_with_docker():
     assert output_file.readline() == "[[ 58  64]\n"
     assert output_file.readline() == " [139 154]]"
     output_file.close()
+
+
+@pytest.mark.integration
+def test_python3_streaming():
+    """
+    Tests python3 with output streaming enabled
+    """
+    test_dir = "./temp_test_python3_streaming"
+    os.makedirs(f"{test_dir}", exist_ok=True)
+    test_script_path = f"{test_dir}/test_script.py"
+
+    with open(test_script_path, "w") as test_script_file:
+        script = """
+import datetime
+import time
+print("attempting to print timestamps every second")
+for _ in range(5):
+    message = datetime.datetime.utcnow().isoformat() + "Z"
+    print(message)
+    time.sleep(1)
+with open("./output/output.txt", "w") as f:
+    f.write("Success")
+            """
+        test_script_file.write(script)
+
+    # Run with captured stdout
+    captured_stdout = io.StringIO()
+    sys.stdout = captured_stdout
+    toolchest.python3(
+        script=test_script_path,
+        output_path=f"{test_dir}/",
+        instance_type=InstanceType.COMPUTE_2,
+        streaming_enabled=True,
+    )
+    # Reset stdout capture
+    sys.stdout = sys.__stdout__
+
+    # Verify toolchest.python3() output files
+    with open(f"{test_dir}/output.txt", "r") as output_file:
+        assert output_file.readline() == "Success"
+
+    # Check printed stdout
+    number_of_expected_following_timestamps = 0
+    last_timestamp = None
+    for line in captured_stdout.getvalue().splitlines():
+        if line == "attempting to print timestamps every second":
+            number_of_expected_following_timestamps = 5
+        elif number_of_expected_following_timestamps > 0:
+            timestamp = datetime.datetime.fromisoformat(line.rstrip("Z"))
+            print(timestamp)
+            # Assert that each pair of timestamps occurred within 2 seconds
+            if last_timestamp:
+                time_diff = timestamp - last_timestamp
+                assert time_diff < datetime.timedelta(seconds=2)
+            last_timestamp = timestamp
+            number_of_expected_following_timestamps -= 1
```

Comment:
This is a bit complicated for "check that this output was streamed". Is there a way that you can just make this check against a constant?

Reply:
Yup, by simplifying the script; refactored